sign.go

```diff
@@ -286,9 +282,27 @@ func signAttributes(attrs []attribute, pkey crypto.PrivateKey, digestAlg crypto.
 	h.Write(attrBytes)
 	hash := h.Sum(nil)
 
+	// dsa doesn't implement crypto.Signer so we make a special case
```

Comment:
Gross. Is there an upstream issue I can look at know why this is the case?

Should we define our own DSA interface that implements `crypto.Signer`? (can be a separate issue)

Reply:
I though about implementing our own DSA type with that interface, but since it's only used in one place, I don't think it's worthwhile.

Here's the upstream: https://github.com/golang/go/issues/27889

Reply:
Thanks!